Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -3105,6 +3105,22 @@ - (TSInteraction *)interactionAtIndexPath:(NSIndexPath *)indexPath {
 
 #pragma mark - Audio
 
+- (void)requestRecordingVoiceMemo
+{
+    OWSAssert([NSThread isMainThread]);
+    [[AVAudioSession sharedInstance] requestRecordPermission:^(BOOL granted) {
```

Comment:
Probably should use weakSelf in case messages view is dismissed while "recording permission" alert is presented, e.g. if app receives an "export with Signal" event.